repositories/patches/rmw_fastrtps-fix-typesupport-conditions-bug.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/rmw_fastrtps_dynamic_cpp/src/type_support_common.cpp b/rmw_fastrtps_dynamic_cpp/src/type_support_common.cpp
+index d602069..62e4362 100644
+--- a/rmw_fastrtps_dynamic_cpp/src/type_support_common.cpp
++++ b/rmw_fastrtps_dynamic_cpp/src/type_support_common.cpp
+@@ -25,12 +25,13 @@
+ bool
+ using_introspection_c_typesupport(const char * typesupport_identifier)
+ {
+-  return typesupport_identifier == rosidl_typesupport_introspection_c__identifier;
++  return !std::string(typesupport_identifier)
++         .compare(rosidl_typesupport_introspection_c__identifier);
```

Comment:
Would std::strcmp work if you would cast `rosidl_typesupport_introspection_c__identifier` to `char*`?

Reply:
strcmp works, I just wanted to use the same method as the patch for cyclonedds to be consistent

Reply:
As pointed out in https://github.com/ros2/rmw_cyclonedds/pull/320, std::strcmp is a bit of an optimization and will avoid an extra mem. alloc. (Eventually we should also refactor the rwm_cyclonedds patch, but that's for a follow-up.)